source/adapters/native_cpu/enqueue.cpp

```diff
@@ -108,10 +125,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   const size_t numParallelThreads = tp.num_threads();
   hKernel->updateMemPool(numParallelThreads);
   std::vector<std::future<void>> futures;
-  std::vector<std::function<void(size_t, ur_kernel_handle_t_)>> groups;
-  auto numWG0 = ndr.GlobalSize[0] / ndr.LocalSize[0];
-  auto numWG1 = ndr.GlobalSize[1] / ndr.LocalSize[1];
-  auto numWG2 = ndr.GlobalSize[2] / ndr.LocalSize[2];
+  std::vector<LaunchInfoLocalArgs> groups;
+  const auto numWG0 = ndr.GlobalSize[0] / ndr.LocalSize[0];
+  const auto numWG1 = ndr.GlobalSize[1] / ndr.LocalSize[1];
+  const auto numWG2 = ndr.GlobalSize[2] / ndr.LocalSize[2];
```

Comment:
Not directly related to this PR, but do we require global size to be a multiple of local size (for each dimension)? If so, an assert would be useful. If not, this looks suspicious. Either way, no need to change that as part of this PR unless you want to, if changes are needed we can do that in a new PR.